app/models/solution.rb

```diff
@@ -15,47 +15,42 @@ class Solution < ApplicationRecord
 
   delegate :auto_approve?, to: :exercise
 
-  def self.completed
-    where.not(completed_at: nil)
-  end
+  scope :core, -> { joins(:exercise).merge(Exercise.core) }
+  scope :side, -> { joins(:exercise).merge(Exercise.side) }
 
-  def self.not_completed
-    where(completed_at: nil)
-  end
+  scope :completed, -> { where.not(completed_at: nil) }
+  scope :not_completed, -> { where(completed_at: nil) }
 
-  def self.published
-    where.not(published_at: nil)
-  end
+  scope :published, -> { where.not(published_at: nil) }
 
-  def self.started
-    where("EXISTS(SELECT NULL FROM iterations WHERE iterations.solution_id = solutions.id)
+  scope :legacy, -> { where("solutions.created_at < ?", Exercism::V2_MIGRATED_AT) }
+  scope :not_legacy, -> { where("solutions.created_at >= ?", Exercism::V2_MIGRATED_AT) }
```

Comment:
Just  in case it matters; ignore me if it's irrelevant.
Is a new iteration an _updated solution_? If so, should this scope be  `updated_at > V2`  ?
So that new iterations on legacy solutions are not legacy.

Reply:
Nice question. So any solution that was *started* before the legacy point is considered legacy. Any solution that wasn't started is considered not_legacy. 

In v2, solutions are created when they are unlocked, not submitted. At the migration point, a lot of solutions were unlocked (so created). If they were unlocked at that point, then they weren't started at that point as they didn't already exist in the db.

This bit is a bit weird :)